test/common/router/router_test.cc

```diff
@@ -1129,20 +1129,25 @@ TEST_F(RouterTest, UpstreamPerTryTimeout) {
   EXPECT_TRUE(verifyHostUpstreamStats(0, 1));
 }
 
-TEST_F(RouterTest, PerTryTimeoutWithNoUpstreamHost) {
+// Ensures that the per try callback is not set until the stream becomes available.
+TEST_F(RouterTest, UpstreamPerTryTimeoutExcludesNewStream) {
   NiceMock<Http::MockStreamEncoder> encoder;
   Http::StreamDecoder* response_decoder = nullptr;
+  Http::ConnectionPool::Callbacks* pool_callbacks;
   EXPECT_CALL(cm_.conn_pool_, newStream(_, _))
       .WillOnce(Invoke([&](Http::StreamDecoder& decoder, Http::ConnectionPool::Callbacks& callbacks)
                            -> Http::ConnectionPool::Cancellable* {
         response_decoder = &decoder;
-        // simulate connect timeout, do not call callbacks.onPoolReady(...)
-        UNREFERENCED_PARAMETER(callbacks);
-        return &cancellable_;
+        pool_callbacks = &callbacks;
+        return nullptr;
+      }));
+  EXPECT_CALL(callbacks_.stream_info_, onUpstreamHostSelected(_))
+      .WillOnce(Invoke([&](const Upstream::HostDescriptionConstSharedPtr host) -> void {
+        EXPECT_EQ(host_address_, host->address());
       }));
 
+  per_try_timeout_ = new Event::MockTimer(&callbacks_.dispatcher_);
   expectResponseTimerCreate();
-  expectPerTryTimerCreate();
```

Comment:
How is it that other tests don't need to be fixed as well? Is it because we don't use InSequence on these tests?

Per @htuch with the boilperplate comment, can you open a tech debt issue to help clean up these tests in terms of reducing boilerplate and also adding InSequence to all of them?